### ISSUE TITLE
/recent_views endpoint query uses non-indexed fields

### ISSUE THREAD

**Issue by paoliniluis** [metabase/metabase#27216]
**Describe the bug**
If we see the card query, it uses executor_id and context which are not indexed
https://github.com/metabase/metabase/blob/d736678ba6b510a793c732a276faa0ac1cf07a21/src/metabase/api/activity.clj#L176, so it does a full table scan

**Logs**
just check with
```
SELECT
    tablename,
    indexname,
    indexdef
FROM
    pg_indexes
WHERE
    schemaname = 'public'
AND tablename = 'query_execution'
ORDER BY
    tablename,
    indexname;
```

**To Reproduce**
Just hit the endpoint and see the query that's being run on the DB

**Expected behavior**
Fast response

**Screenshots**
NA

**Information about your Metabase Installation:**
- Your databases: Postgres
- Metabase version: 45.1
- Metabase hosting environment: Docker
- Metabase internal database: PG 15.1

**Severity**
P2

**Additional context**
Reported by a customer

Query plan
```
EXPLAIN ANALYZE SELECT min("executor_id") AS "user_id", "query_execution"."card_id" AS "model_id", count(*) AS "cnt", max("started_at") AS "max_ts" FROM "query_execution" LEFT JOIN "card_bookmark" "bm" ON ("bm"."user_id" = 1 AND "query_execution"."card_id" = "bm"."card_id") WHERE ("executor_id" = 1 AND "context" = 'question' AND "bm"."id" IS NULL) GROUP BY "query_execution"."card_id", "context" ORDER BY "max_ts" DESC LIMIT 8
```

```
Limit  (cost=23.78..23.79 rows=1 width=106) (actual time=0.074..0.090 rows=1 loops=1)
  ->  Sort  (cost=23.78..23.79 rows=1 width=106) (actual time=0.071..0.083 rows=1 loops=1)
        Sort Key: (max(query_execution.started_at)) DESC
        Sort Method: quicksort  Memory: 25kB
        ->  GroupAggregate  (cost=23.74..23.77 rows=1 width=106) (actual time=0.049..0.060 rows=1 loops=1)
"              Group Key: query_execution.card_id, query_execution.context"
              ->  Sort  (cost=23.74..23.75 rows=1 width=98) (actual time=0.041..0.049 rows=1 loops=1)
                    Sort Key: query_execution.card_id
                    Sort Method: quicksort  Memory: 25kB
                    ->  Nested Loop Left Join  (cost=0.15..23.73 rows=1 width=98) (actual time=0.018..0.026 rows=1 loops=1)
                          Filter: (bm.id IS NULL)
                          ->  Seq Scan on query_execution  (cost=0.00..15.55 rows=1 width=98) (actual time=0.007..0.010 rows=1 loops=1)
                                Filter: ((executor_id = 1) AND ((context)::text = 'question'::text))
                                Rows Removed by Filter: 5
                          ->  Index Scan using unique_card_bookmark_user_id_card_id on card_bookmark bm  (cost=0.15..8.17 rows=1 width=8) (actual time=0.006..0.007 rows=0 loops=1)
                                Index Cond: ((user_id = 1) AND (card_id = query_execution.card_id))
Planning Time: 0.949 ms
Execution Time: 0.149 ms
```

check the "Seq Scan on query_execution" which could explode the DB on massive tables

**Comment by joekohlsdorf** [comment]
@tsmacdonald Please reopen the issue, I added the two indexes manually and they don't help.
Here is a query plan on Postgres after adding them.

The query planner doesn't use the index because from statistics it knows that most rows fulfill the `"context" = 'question'` 
condition so full table scan is more efficient.

In my opinion the problem is how the query of this feature is designed.

```
Limit  (cost=648963.31..648963.33 rows=8 width=31) (actual time=59356.846..59419.515 rows=8 loops=1)
  ->  Sort  (cost=648963.31..649066.99 rows=41472 width=31) (actual time=59356.845..59419.513 rows=8 loops=1)
        Sort Key: (max(query_execution.started_at)) DESC
        Sort Method: top-N heapsort  Memory: 26kB
        ->  Finalize GroupAggregate  (cost=640660.27..648133.87 rows=41472 width=31) (actual time=54609.414..59413.442 rows=34317 loops=1)
              Group Key: query_execution.card_id, query_execution.context
              ->  Gather Merge  (cost=640660.27..647090.80 rows=50268 width=31) (actual time=54609.407..59383.101 rows=81592 loops=1)
                    Workers Planned: 2
                    Workers Launched: 2
                    ->  Partial GroupAggregate  (cost=639660.25..640288.60 rows=25134 width=31) (actual time=54555.540..59029.242 rows=27197 loops=3)
                          Group Key: query_execution.card_id, query_execution.context
                          ->  Sort  (cost=639660.25..639723.08 rows=25134 width=23) (actual time=54555.518..58243.380 rows=3981755 loops=3)
                                Sort Key: query_execution.card_id
                                Sort Method: external merge  Disk: 164096kB
                                Worker 0:  Sort Method: external merge  Disk: 163752kB
                                Worker 1:  Sort Method: external merge  Disk: 163288kB
                                ->  Hash Left Join  (cost=14.47..637823.29 rows=25134 width=23) (actual time=1.054..41053.746 rows=3981755 loops=3)
                                      Hash Cond: (query_execution.card_id = bm.card_id)
                                      Filter: (bm.id IS NULL)
                                      ->  Parallel Seq Scan on query_execution  (cost=0.00..624603.07 rows=5026762 width=23) (actual time=0.995..40198.378 rows=3981755 loops=3)
                                            Filter: ((context)::text = 'question'::text)
                                            Rows Removed by Filter: 5689011
                                      ->  Hash  (cost=14.37..14.37 rows=8 width=8) (actual time=0.023..0.024 rows=0 loops=3)
                                            Buckets: 1024  Batches: 1  Memory Usage: 8kB
                                            ->  Bitmap Heap Scan on card_bookmark bm  (cost=4.21..14.37 rows=8 width=8) (actual time=0.022..0.023 rows=0 loops=3)
                                                  Recheck Cond: (user_id = 10117)
                                                  ->  Bitmap Index Scan on idx_card_bookmark_user_id  (cost=0.00..4.21 rows=8 width=0) (actual time=0.021..0.021 rows=0 loops=3)
                                                        Index Cond: (user_id = 10117)
Planning Time: 0.296 ms
Execution Time: 59427.691 ms
```

**Comment by camsaul** [comment]
If the indexes we added in #27606 are not solving our problems, we should remove them, because these tables get written to extremely often, and having to update these indexes is going to negatively impact write thruput for them.

**Comment by qnkhuat** [comment]
`GET /api/logs/query_execution/:yyyy-mm` is also doing a full table scan on query execution.

**Comment by paoliniluis** [comment]
This issue caused downtime today in a customer with 30M rows on the query execution table and 10M on the view log one using MySQL

**Comment by noahmoss** [comment]
(Closed as this has now been backported for the upcoming minor release)

**Comment by terite** [comment]
I experienced this issue today running `v0.47.2`. Our `query_execution` table had approximately 6.3 million rows in it, and we observed the relevant query taking approximately 9 minutes to complete. I'd prefer to keep audit history, but I removed everything older than 6 months to get our Metabase instance usable again

I'd like to second what @joekohlsdorf mentioned earlier, this seems like an issue with the design of the feature.

I also looked at [MB_AUDIT_MAX_RETENTION_DAYS](https://www.metabase.com/docs/latest/configuring-metabase/environment-variables#mb_audit_max_retention_days), but that's listed as only working on Pro and Enterprise plans, which this particular Metabase installation is not.
